text/0328-polyglot-assert.md

```diff
@@ -0,0 +1,483 @@
+`Foo::Bar` contains the specified properties -
+
+```ts
+// In typescript
+assert.assertHasResource('Foo::Bar', {
+  Foo: 'Bar',
+  Baz: 5,
+  Qux: [ 'Waldo', 'Fred' ],
+});
+```
+
+```java
+// In Java, using text blocks and Gson
+import com.google.gson.Gson;
+
+String json = """
+  {
+    "Foo": "Bar",
+    "Baz": 5,
+    "Qux": [ "Waldo", "Fred" ],
+  } """;
+
+Map expected = new Gson().fromJson(json, Map.class);
+assert.assertHasResource("Foo::Bar", expected);
+```
+
+```py
+# In Python
+import json
+
+expected = """
+  {
+    "Foo": "Bar",
+    "Baz": 5,
+    "Qux": [ "Waldo", "Fred" ],
+  } """;
+
+assertion.assertHasResource('Foo::Bar', json.loads(expected))
+```
+
+The same method allows asserting the complete definition of the 'Resource'
+which can be used to verify things other sections like `DependsOn`, `Metadata`,
+`DeletionProperty`, etc.
+
+```ts
+// In typescript
+assert.assertHasResource('Foo::Bar', {
+  Properties: { Foo: 'Bar' },
+  DependsOn: [ 'Waldo', 'Fred' ],
+}, {
+  part: ResourcePartBeta1.COMPLETE,
+});
+```
+
+```java
+// In Java, using text blocks and Gson
+import com.google.gson.Gson;
+
+String json = """
+  {
+    "Properties": { "Foo": "Bar" },
+    "DependsOn": [ "Waldo", "Fred" ],
+  } """;
+
+Map expected = new Gson().fromJson(json, Map.class);
+assert.assertHasResource("Foo::Bar", expected,
+  new AssertResourceOptionsBeta1.Builder().part(ResourcePartBeta1.COMPLETE).build());
+```
+
+```py
+# In Python
+import json
+
+expected = """
+  {
+    "Properties": { "Foo": "Bar" },
+    "DependsOn": [ "Waldo", "Fred" ],
+  } """;
+
+assertion.assertHasResource('Foo::Bar', json.loads(expected),
+  assertion.AssertResourceOptionsBeta1(part=ResourcePartBeta1.COMPLETE))
+```
+
+## FAQ
+
+### What are we launching today?
+
+Today, we are launching a new jsii module called `@aws-cdk/assertions`, that allows
+CDK users to run assertions on the synthesized CloudFormation template.
+
+This module is available to users in all CDK supported languages.
+
+### Why should I use this feature?
+
+You should use this feature if you would like to ensure that your CDK constructs
+and CDK applications render expected CloudFormation resources and properties.
+
+### How is this different from the existing `assert` module?
+
+The two modules provide the same set of features. However, the existing module
+was only available to javascript users.
+The new version launched today is available to users of all CDK supported
+languages available today and any that we will support in the future.
+
+### How do I get started?
+
+This module is available in the languages as per the table below -
+
+| Language              | Pkg Manager | Package                                       |
+| --------------------- | ------------|---------------------------------------------- |
+| typescript/javascript | npm         | `@aws-cdk/assertions`                         |
+| Java                  | Maven       | `software.amazon.awscdk.assertions`           |
```

Comment:
why? 

Isn't it a valid FAQ that users would expect to see? especially given that, the primary value prop of this RFC is that the module is available in all languages.

Reply:
Of course it is but just feels like it's a detail we don't have to expand on in the RFC since it's derived from the submodule name and there isn't any control here.